instance/management/commands/migrate_swift_to_s3.py

```diff
@@ -0,0 +1,135 @@
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Migrate SWIFT assets to S3.
+One-time script.
+"""
+
+# Imports #####################################################################
+
+import re
+import subprocess
+import logging
+
+from django.core.management.base import BaseCommand
+
+from instance.models.openedx_instance import OpenEdXInstance
+import swiftclient
+
+LOG = logging.getLogger(__name__)
+
+# Classes #####################################################################
+
+
+class Command(BaseCommand):
+    """
+    Migrate SWIFT to S3.
+    """
+    help = (
+        'Migrate all servers\' data from SWIFT to S3.'
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.options = {}
+        self.retried = {}
+
+    def _get_swift_connection(self, instance):
+        return swiftclient.Connection(
+            user=instance.swift_openstack_user,
+            key=instance.swift_openstack_password,
+            authurl=instance.swift_openstack_auth_url,
+            tenant_name=instance.swift_openstack_tenant,
+            auth_version='2',
+            os_options={'region_name': instance.swift_openstack_region}
+        )
+
+    def _create_rclone_config(self, access_key, secret_key):
+        """
+        Creates rclone config, to ease the process of copying
+        """
+        rclone_config = "rclone config create ocim-s3 s3 access_key_id %s secret_access_key %s" % (access_key, secret_key)
+        print("Would run this command: %s.       ← FIXME test it and run it" % rclone_config)
+        return
+
+        subprocess.Popen(rclone_config % (access_key, secret_key), shell=True)
+
+
+    def _copy_subdirectory(self, key_name, new_name, swift_container, s3_bucket):
+        """
+        Copy subdirectory using rclone
+        """
+        LOG.info("%s ---> %s" % (key_name, new_name))
+        command = "rclone copy ocim-swift:%s/%s ocim-s3:%s/%s" % (swift_container, key_name, s3_bucket, new_name)
+        print("Would run this command: %s.       ← FIXME test it and run it" % command)
+        return
+        subprocess.Popen(command, shell=True)
+
+    def _migrate_swift_to_s3(self, instance):
+        # FIXME Will these 2 values already be stored in the instance? If not, where to get them from? (Or should we set them and save?)
+        # FIXME and in particular: will the S3 region match the SWIFT region? (some clients don't want their data in a particular region; see comments on ticket)
```

Comment:
No, there is no support for specifying region now, we will have to skip cloudera this time.

Reply:
How much work would it be to hard-code the region to Paris?  Our instances are hosted in France, so it would make sense to have the object storage in France as well, and that would also work for Cloudera.

Reply:
Shouldn't be much work to hardcode that.

Reply:
Then let's do this please.  It would allow us to completely drop Swift support.  Swift has required significant maintenance to keep working in the past, because it's not tested by upstream, and I would like to get rid of that maintenance effort.  Moreover, we already had at least two customers requiring storage within the EU, so storing the data in Paris by default seems like a good idea.

Reply:
Would that mean: the same S3 user and password for all instances?

Reply:
(Yes, same user/pass).

However, there's a problem: if I try to upload things to a bucket without knowing the region, I get errors like `BucketRegionError: incorrect region, the bucket is not in 'eu-central-1' region`. This means that either we must store the region of each instance's S3 (same as we store SWIFT's region) or we change some default in AWS web so that everything goes to Paris, *and* then we migrate the existing buckets from non-Paris to Paris so that all of them be in Paris (and then we can't have buckets in US).

I think we should store the region, but we need a new field for that, near `s3_bucket_name`; I can create a ticket.